innersource-license.md

```diff
@@ -0,0 +1,83 @@
+
+## Problem
+
+When two companies within a group want to share code with each other, they need an agreement about the terms and often a legal contract. Creating such agreements on a per project basis takes effort and creates a barrier for sharing. i.e. a team/company might decide not to share their source code with another company in the group because it seems complicated.
+
+Barriers for sharing can lead to silos and duplication of effort in rebuilding similar solutions in multiple parts of the group.
+
+At the time of sharing the source code, it can not be reliably predicted what the value of sharing will be. Hence if the activity of sharing has any significant cost, the companies are unlikely to do it, as they are concerned about the return on investment.
+
+## Context
+
+- A large group with many companies (subsidiaries) that want to share code. (when the group gets larger, the value of this pattern increases)
+- These companies are legally independent units.
+- Multiple of these companies/subsidiaries are developing software, and are using services of the other companies. They have a motivation to contribute to each other’s source code.
+- Certain size of developers (crowdsourcing and ecosystem effects of InnerSource)
+
+## Forces
+
+- **Level of effort** required to write formal agreements, especially if they need to take into account technical, legal, and business perspectives.
+- Enterprises have many **internal regulations,** any agreements have to comply with these regulations, e.g. security, privacy, procurement processes, etc. The volume of regulations can make it difficult to assess whether sharing software between two companies is compliant with these regulations, especially when there is no standard procedure.
+- If any of the companies in the group has a **business model** that depends on proprietary code and accounting of licensing fees within the group
+- **Company culture** that isn’t used to sharing code
+- Freedom over using the software leads to competition, and spread of ownership
+
+## Solutions
+
+Creating an **InnerSource License** customized to the needs of the group in question (and their companies). This license needs to be generic enough to be applied to the most important inter-company relationships.
+
+It is important to write the InnerSource License such that it truly allows for OpenSource-like collaborations across organizational boundaries. Therefore the 4 freedoms of free software should be integrated into the license.
+
+The License is written as a formal legal document, and can be used as part of contracts between the subsidiaries to govern the code sharing agreements.
+
+## Resulting Context
+
+With the InnerSource License, we have a tool to share code between companies within our group.
```

Comment:
Can we beef that up somehow? 

- What are thing people should be aware of when applying this pattern?
- What were positive side effects observed?
- What were negative side effects observed? (I always feel that is the greatest part of the resulting context)

Reply:
I think what you are looking for is mostly described in **Known Instances**. There we tried to describe the observations that are specific to DB Systel. Also as they have barely rolled this InnerSource License concept out, the experience is not that hardened yet. 

Maybe we can review this **Resulting Context** again in a couple of months or together with other companies that have applied this pattern too?

I would not change this for now.

@maxcapraro please close this conversation if you are fine with leaving this section as is.

Reply:
My guts feeling would be to still synthesize it and already add it to the resulting context. I'd like to hear @lenucksi's / other's opinion on that, though.

Reply:
What @spier states here is actually what I would more or less add to the resulting context.
- We could very thinly paraphrase the Known Instances, point there and also state that research is still out on specific,reliable resultung context results.

Reply:
I have beefed up the resulting context just a little bit, following @lenucksi suggestion. Resolving this thread.